api/controllers/EAppReferenceController.js

```diff
@@ -19,25 +19,76 @@ const EAppReferenceController = {
             user_data: userData,
             userRef: req.session.eApp.userRef,
             maxReferenceLength: MAX_CHAR_LENGTH,
-            inputError: false,
+            referenceErrors: [],
         });
     },
 
     addReferenceToSession(req, res) {
         const userRef = req.body['user-reference'];
+        const illegalCharacters =
+            HelperService.checkForIllegalCharacters(userRef);
+        const isOverCharLimit = userRef.length > MAX_CHAR_LENGTH;
 
-        if (userRef.length > MAX_CHAR_LENGTH) {
+        if (illegalCharacters.error) {
+            return res.serverError(illegalCharacters.error);
+        }
+
+        EAppReferenceController._checkReferenceForErrors({
+            req,
+            isOverCharLimit,
+            illegalCharacters,
+        });
+
+        const referenceErrors = req.flash('referenceErrors');
+
+        if (isOverCharLimit || illegalCharacters.exist) {
             return res.view('eApostilles/additionalReference.ejs', {
                 user_data: HelperService.getUserData(req, res),
                 userRef: '',
                 maxReferenceLength: MAX_CHAR_LENGTH,
-                inputError: true,
+                referenceErrors,
             });
         }
 
         req.session.eApp.userRef = userRef;
         return res.redirect('/check-uploaded-documents');
     },
+
+    _checkReferenceForErrors({ req, isOverCharLimit, illegalCharacters }) {
+        const errorMessages = {
+            illegalCharacter: {
+                title: 'There is a problem with your reference',
+                text: 'The reference cannot use the following characters:',
```

Comment:
As you've now got a restricted character set you may want to consider including the accepted characters in the error message to give the user an idea of what value they can submit rather than just highlighting the characters that are causing the error.  Not a show stopper but maybe one for Liz to opine on.